src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV1.cs

```diff
@@ -56,34 +56,22 @@ public class PublishPipelineArtifactTaskV1 : PipelineArtifactTaskPluginBaseV1
         protected override string TargetPath => "path";
 
         private static readonly Regex jobIdentifierRgx = new Regex("[^a-zA-Z0-9 - .]", RegexOptions.Compiled | RegexOptions.CultureInvariant);
+        private static readonly string pipelineType = "pipeline";
+        private static readonly string fileShareType = "filepath";
 
         protected override async Task ProcessCommandInternalAsync(
             AgentTaskPluginExecutionContext context, 
             CancellationToken token)
         {
             string artifactName = context.GetInput(ArtifactEventProperties.ArtifactName, required: false);
             string targetPath = context.GetInput(TargetPath, required: true);
+            string artifactType = context.GetInput(ArtifactEventProperties.ArtifactType, required: true);
+            artifactType = artifactType.ToLower();
+
             string defaultWorkingDirectory = context.Variables.GetValueOrDefault("system.defaultworkingdirectory").Value;
 
             targetPath = Path.IsPathFullyQualified(targetPath) ? targetPath : Path.GetFullPath(Path.Combine(defaultWorkingDirectory, targetPath));
-
-            string hostType = context.Variables.GetValueOrDefault(WellKnownDistributedTaskVariables.HostType)?.Value; 
-            if (!string.Equals(hostType, "Build", StringComparison.OrdinalIgnoreCase)) {
-                throw new InvalidOperationException(
-                    StringUtil.Loc("CannotUploadFromCurrentEnvironment", hostType ?? string.Empty)); 
-            }
-
-            if (String.IsNullOrWhiteSpace(artifactName))
-            {
-                string jobIdentifier = context.Variables.GetValueOrDefault(WellKnownDistributedTaskVariables.JobIdentifier).Value;
-                var normalizedJobIdentifier = NormalizeJobIdentifier(jobIdentifier);
-                artifactName = normalizedJobIdentifier;
-            }
-
-            if(!PipelineArtifactPathHelper.IsValidArtifactName(artifactName)) {
-                throw new ArgumentException(StringUtil.Loc("ArtifactNameIsNotValid", artifactName));
-            }
-
+   
             // Project ID
             Guid projectId = new Guid(context.Variables.GetValueOrDefault(BuildVariables.TeamProjectId)?.Value ?? Guid.Empty.ToString());
```

Comment:
>ToString [](start = 125, length = 8)

Why do we need ToString here?

Just do the condition on the whether the value is there, if it's there parse the GUID, otherwise give empty GUID

Reply:
It has to be a string. What it is doing is the create the GUID from a string. if the string is empty, then create a Guid using an emtpy Guid string.

Reply:
You can do this instead:

teamProjectId = context.Variables.GetValueOrDefault(BuildVariables.TeamProjectId);
            Guid projectId = teamProjectId == null ? new Guid(teamProjectId) : Guid.Empty;


---
In reply to: [313602556](https://github.com/microsoft/azure-pipelines-agent/pull/2385#discussion_r313602556) [](ancestors = 313602556)

Reply:
that way you don't need to do an extra ToString call

---
In reply to: [313636009](https://github.com/microsoft/azure-pipelines-agent/pull/2385#discussion_r313636009) [](ancestors = 313636009,313602556)